packages/core/src/ui/Base/TwitterEmbed.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useCallback, useState, useRef, RefObject } from "react";
+
+const twitterWidgetJs = "https://platform.twitter.com/widgets.js";
+const message = "Loading tweet...";
+
+interface TweetConfig {
+  theme: string;
+}
+
+type TweetState = {
+  isLoading: boolean;
+  message: string | null;
+};
+
+declare global {
+  interface Window {
+    twttr: {
+      widgets: {
+        createTweet: (
+          id: string,
+          ref: RefObject<HTMLDivElement>,
+          options: TweetConfig
+        ) => Promise<any>; // TODO type
+        load: (ref: RefObject<HTMLDivElement>) => void;
+      };
+    };
+  }
+}
+
+export default function TwitterEmbed({ url, ...props }) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [state, setState] = useState<TweetState>({
+    isLoading: true,
+    message: message,
+  });
+
+  const tweetId = url.split("status/").pop();
+
+  const renderTweet = useCallback(() => {
```

Comment:
What's the benefit of using `useCallback` here?

Reply:
I get the following if not using a useCallback: 

The 'renderTweet' function makes the dependencies of useEffect Hook change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'renderTweet' in its own useCallback() Hook.

Reply:
Yes, I know, but the `renderTweet` function doesn't even have to be a dependency of `useEffect`. It only makes the code less readable in this case and is used here like a patch, because sth is showing an error if not done otherwise...
The reason why in this case `renderTweet` has to be listed as a dependency of `useEffect` is that 1) it's called inside the `useEffect` callback AND 2) it's not a pure function, i.e. it has references to props (or state) which we want to trigger the `useEffect`. Instead, we could be passing these props (or state) directly to `useEffect`. 
And if `renderTweet` was a pure* function, i.e. it would be explicitly passed values it needs as arguments (instead of accessing them from the upper scope) yes, on each render another instance of it would be created (which is not a problem in this case) but it would have exactly the same effect when run with the same arguments. And you wouldn't have to pass it then as a dependency of `useEffect.` 

Another way of doing it is what the error suggests to you in the first place: moving this function inside the useEffect callback.

*Note: pure but not in a very strict sense, because it will still have references to `window`, but it the same on each render, so doesn't make a difference.

Let me know if it's still not clear, I didn't want to solve it for you 😛